### DISCUSSION TITLE
JavaMelody JDBC connection graphs

### DISCUSSION THREAD

**Discussion by asutoshv** [javamelody/javamelody#983] [Q&A] [answered]
Hello All, 
I'm reviewing graphs of active and used jdbc connections for an application. Active connections are 773 m ; and used connections are 3 k. I'm not sure what m and k means here? 
Can someone help please ?

![jdbc](https://user-images.githubusercontent.com/77450305/104616352-1734b700-5682-11eb-8546-acc84c38897b.jpg)

**Comment by evernat** [comment, marked as answer]
Hi and welcome in the disscussions.
This is explained in the online help, [in English](http://javamelody.org/demo/monitoring?resource=help/help.html) for example.
Here is an extract for you:
> The units in the charts are chosen based on the values: "m" is for milli (1 / 1000), "k" is for kilo/thousands, "M" is for mega/millions, "G" is for giga/billions and "u" is for micro (1 / 1000000).

As you can see in the graphs, for that day, the mean of active jdbc connections (jdbc connections executing requests) is below 1 and the maximum is 9. It seems reasonnable in general and not very high.

And the mean of used jdbc connections (jdbc connections which were opened by the application and not yet closed, neither released to the jdbc pool) is 3000 and the maximum is 4000 with always increasing values until the server is stopped. That's a lot compared to usual values in applications. I do not know the specificities  of the jdbc connections in this application, but it really seems like a leak of jdbc connections. This leak should probably be fixed. You can look at the "Opened jdbc connections" link in the middle of the monitoring report if it helps to find where those jdbc connections were opened and to fix this leak of jdbc connections.

**Reply by asutoshv** [reply]
Thank you very much for the information. Its spot on.
